azext_iot/tests/test_iot_ext_int.py

```diff
@@ -1165,76 +1170,31 @@ def test_hub_monitor_event_all(self):
         client = iot_hub_service_factory(cli_ctx)
         device_count = 10
 
-        # Event monitor should throw runtime exception in case of error
-        self.cmd('iot hub monitor-events -t 10 -y -p all --login {}'.format(LIVE_HUB_CS + 'zzz'), expect_failure=True)
+        # Test with invalid connection string
+        self.cmd('iot hub monitor-events -t 1 -y --login {}'.format(LIVE_HUB_CS + 'zzz'), expect_failure=True)
 
-        names = self._create_entity_names(devices=device_count)
-        device_ids = names['device_ids']
+        # Create and Simulate Devices
+        device_ids = self._create_entity_names(devices=device_count)['device_ids']
 
         for i in range(device_count):
             self.cmd('iot hub device-identity create -d {} -n {} -g {}'.format(device_ids[i], LIVE_HUB, LIVE_RG),
                      checks=[self.check('deviceId', device_ids[i])])
 
-        for i in range(device_count):
-            execute_onthread(method=iot_simulate_device,
-                             args=[client, device_ids[i], LIVE_HUB, 'complete', 'Ping from event monitor test 1, part a', 5, 1],
-                             max_runs=1)
-
-        self.cmd('iot hub monitor-events -n {} -g {} -t 10 -y -p sys anno app'.format(
-            LIVE_HUB, LIVE_RG), checks=None)
+        enqueued_time = datetime.datetime.now().microsecond
 
-        # With connection string
         for i in range(device_count):
-            execute_onthread(method=iot_simulate_device,
```

Comment:
The advantage with iot_simulate_device is you can see the event monitor stability over an elapsed period of time. iot_device_send_message does have a lot more power/flexibility per message though.